x-pack/plugins/security_solution/server/endpoint/routes/resolver/tree/queries/stats.ts

```diff
@@ -121,26 +186,99 @@ export class StatsQuery {
    * @param client used to make requests to Elasticsearch
    * @param nodes an array of unique IDs representing nodes in a resolver graph
    */
-  async search(client: IScopedClusterClient, nodes: NodeID[]): Promise<Record<string, EventStats>> {
+  async search(
+    client: IScopedClusterClient,
+    nodes: NodeID[],
+    alertsClient: AlertsClient | undefined,
+    includeHits: boolean
+  ): Promise<{ eventStats?: Record<string, EventStats>; alertIds?: string[] }> {
     if (nodes.length <= 0) {
       return {};
     }
 
     const esClient = this.isInternalRequest ? client.asInternalUser : client.asCurrentUser;
+    const alertIndex =
+      typeof this.indexPatterns === 'string' ? this.indexPatterns : this.indexPatterns.join(',');
 
-    // leaving unknown here because we don't actually need the hits part of the body
-    const body = await esClient.search({
-      body: this.query(nodes),
-      index: this.indexPatterns,
+    const [body, alertsBody] = await Promise.all([
+      await esClient.search({
+        body: this.query(nodes),
+        index: this.indexPatterns,
+      }),
+      alertsClient
+        ? await alertsClient.find(this.alertStatsQuery(nodes, alertIndex, includeHits))
+        : { hits: { hits: [] } },
+    ]);
+    // @ts-expect-error declare aggegations type explicitly
+    const eventAggs: CategoriesAgg[] = body.aggregations?.ids?.buckets ?? [];
+    // @ts-expect-error declare aggegations type explicitly
+    const alertAggs: AggBucket[] = alertsBody.aggregations?.ids?.buckets ?? [];
+    const eventsWithAggs = new Set([
+      ...eventAggs.map((agg) => agg.key),
+      ...alertAggs.map((agg) => agg.key),
+    ]);
+    const alertsAggsMap = new Map(alertAggs.map(({ key, doc_count: docCount }) => [key, docCount]));
+    const eventAggsMap = new Map<string, EventStats>(
+      eventAggs.map(({ key, doc_count: docCount, categories }): [string, EventStats] => [
+        key,
+        {
+          ...StatsQuery.getEventStats({ key, doc_count: docCount, categories }),
+        },
+      ])
+    );
+    const alertIdsRaw: Array<string | undefined> = alertsBody.hits.hits.map((hit) => {
+      return hit._source && hit._source['kibana.alert.uuid'];
```

Comment:
I think there's a constant for `kibana.alert.uuid` in the common folder. I would use that here considering the alert field naming changes we've done in the past